applications/StructuralMechanicsApplication/custom_elements/nodal_concentrated_element.cpp

```diff
@@ -363,126 +830,142 @@ void NodalConcentratedElement::CalculateDampingMatrix(
 {
     KRATOS_TRY;
 
-    //0.-Initialize the DampingMatrix:
-    const unsigned int dimension = GetGeometry().WorkingSpaceDimension();
+    // The domain size
+    const SizeType dimension = GetGeometry().WorkingSpaceDimension();
 
     // Resizing as needed the LHS
-    const unsigned int system_size = dimension;
+    const SizeType system_size = this->ComputeSizeOfSystem();
 
+    // 0.-Initialize the DampingMatrix:
     rDampingMatrix = ZeroMatrix( system_size, system_size );
 
     //Check, if Rayleigh damping is available; use nodal damping, if not
-    if( mUseRayleighDamping ) {
-        StructuralMechanicsElementUtilities::CalculateRayleighDampingMatrix(
-            *this,
-            rDampingMatrix,
-            rCurrentProcessInfo,
-            system_size);
+    if( mELementalFlags.Is(NodalConcentratedElement::COMPUTE_RAYLEIGH_DAMPING) ) {
+        StructuralMechanicsElementUtilities::CalculateRayleighDampingMatrix(*this, rDampingMatrix, rCurrentProcessInfo, system_size);
     } else {
+        // We get the reference
         const auto& rconst_this = *this;
-        const array_1d<double, 3 >& nodal_damping_ratio = rconst_this.GetValue(NODAL_DAMPING_RATIO);
-        for ( unsigned int j = 0; j < dimension; ++j ) {
-            rDampingMatrix(j, j) += nodal_damping_ratio[j];
+
+        IndexType aux_index = 0;
+        if( mELementalFlags.Is(NodalConcentratedElement::COMPUTE_DAMPING_RATIO) ) {
+            const array_1d<double, 3 >& nodal_damping_ratio = HasProperties() ? (GetProperties().Has(NODAL_DAMPING_RATIO) ? GetProperties().GetValue(NODAL_DAMPING_RATIO) : rconst_this.GetValue(NODAL_DAMPING_RATIO)) : rconst_this.GetValue(NODAL_DAMPING_RATIO);
+            for ( IndexType j = 0; j < dimension; ++j )
+                rDampingMatrix(j, j) += nodal_damping_ratio[j];
+
+            aux_index += dimension;
+        }
+        if( mELementalFlags.Is(NodalConcentratedElement::COMPUTE_ROTATIONAL_DAMPING_RATIO) ) {
+            const array_1d<double, 3 >& nodal_rotational_damping_ratio = HasProperties() ? (GetProperties().Has(NODAL_ROTATIONAL_DAMPING_RATIO) ? GetProperties().GetValue(NODAL_ROTATIONAL_DAMPING_RATIO) : rconst_this.GetValue(NODAL_ROTATIONAL_DAMPING_RATIO)) : rconst_this.GetValue(NODAL_ROTATIONAL_DAMPING_RATIO);
```

Comment:
double ternary operator :see_no_evil:

Reply:
can you make this a separate fct in this cpp, like e.g. I did for the retrieval of the rayleigh factors in the StructMechElementUtils?
This way would already be much more readable